packages/kamino-sdk/src/Kamino.ts

```diff
@@ -2945,6 +2950,46 @@ export class Kamino {
     return totalHoldings.a.div(totalHoldings.b);
   };
 
+  calculateAmountsToBeDepositedWithSwap = async (
+    strategy: PublicKey | StrategyWithAddress,
+    tokenAAmount: Decimal,
+    tokenBAmount: Decimal
+  ): Promise<[Decimal, Decimal, Decimal, Decimal]> => {
+    const { strategy: strategyState } = await this.getStrategyStateIfNotFetched(strategy);
+    let tokenAMint = strategyState.tokenAMint;
+    let tokenBMint = strategyState.tokenBMint;
+
+    const _priceAInB = await this._jupService.getPrice(tokenAMint, tokenBMint);
+    const _priceBInA = await this._jupService.getPrice(tokenBMint, tokenAMint);
+
+    let tokenADecimals = strategyState.tokenAMintDecimals.toNumber();
+    let tokenBDecimals = strategyState.tokenBMintDecimals.toNumber();
+
+    let aAmount = tokenAAmount;
+    let bAmount = tokenBAmount;
+
+    let orcaAmounts = await this.calculateAmountsToBeDeposited(
+      strategy,
+      collToLamportsDecimal(new Decimal(100.0), tokenADecimals)
+    );
+
+    let orcaAmountA = orcaAmounts[0].div(10 ** tokenADecimals);
+    let orcaAmountB = orcaAmounts[1].div(10 ** tokenBDecimals);
+
+    let ratio = orcaAmountA.div(orcaAmountB);
+    ratio = ratio.div(_priceBInA);
+
+    let totalInA = aAmount.add(bAmount.mul(_priceBInA));
```

Comment:
the math seems correct, but variable names are not very intuitive.

```suggestion
    let totalUserDepositInA = aAmount.add(bAmount.mul(_priceBInA));
```